torch/__init__.py

```diff
@@ -2587,6 +2589,43 @@ def get_device_module(device: _Optional[_Union[torch.device, str]] = None):
     return device_module
 
 
+class DeviceGuard:
+    def __init__(self, device_index: builtins.int):
+        self.idx = device_index
+        self.prev_idx = -1
+
+    def __enter__(self):
+        self.prev_idx = torch._C._exchange_device(self.idx)
+
+    def __exit__(self, type: _Any, value: _Any, traceback: _Any):
+        self.idx = torch._C._maybe_exchange_device(self.prev_idx)
+        return False
```

Comment:
`return False` means it allows the exception to be processed normally upon exit from this method. refer to https://docs.python.org/3/reference/datamodel.html#object.__exit__
> If an exception is supplied, and the method wishes to suppress the exception (i.e., prevent it from being propagated), it should return a true value. Otherwise, the exception will be processed normally upon exit from this method.